### PR TITLE
Examples file loader rework

### PR DESCRIPTION
**Pull request by kpal81xd**
- Moves external file loading from config to actual file loading (use `localImport` function)
- Moves flags from config to comment flags in example
- Flattened examples folders into files with preprended exampleName (e.g. `hello-world.example.mjs`)

### Preview

image

### CODE REVIEW

**Comment by mvaligursky** [comment]
Could we also rename the example files so that they're not all called `example.mjs`? Happy if this is done in a separate PR too.

File: examples/src/examples/user-interface/text-typewriter/example.mjs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// @flag WEBGPU_ENABLED
```
Comment (mvaligursky) [inline examples/src/examples/user-interface/text-typewriter/example.mjs:1]: are you using some existing `@flag` functionality? If not, could we give it some better name, perhaps `@example` or something?

**Comment by kpal81xd** [comment]
> Could we also rename the example files so that they're not all called `example.mjs`? Happy if this is done in a separate PR too.

Since there is still a controls file for now I was going to rename the files to `<exampleName>.example.mjs` and `<exampleName.controls.mjs`?

File: examples/src/examples/user-interface/text-typewriter/example.mjs (hunk 1 above)
Comment (kpal81xd) [reply examples/src/examples/user-interface/text-typewriter/example.mjs:1]: No this is custom and the word flag is just arbitrary just to signify its a config flag option so can be called anything

File: examples/src/examples/graphics/ground-fog/shader.frag [hunk 2]
```diff
@@ -0,0 +1,37 @@
+
+            uniform sampler2D uTexture;
```
Comment (willeastcott) [inline examples/src/examples/graphics/ground-fog/shader.frag:2]: Trim all the unnecessary whitespace in this file (and all other shader files that have the same problem).

Comment (kpal81xd) [reply examples/src/examples/graphics/ground-fog/shader.frag:2]: Yep working on it

File: examples/README.md [hunk 3]
```diff
@@ -61,6 +57,20 @@ export { app };
 
 This is the only file that's required to run an example. The code defined in this function is executed each time the example play button is pressed. It takes the example's canvas element from the DOM and usually begins by creating a new PlayCanvas `Application` or `AppBase` using that canvas.
 
+Examples can also contains comments which allow you to define the default configuration for your examples as well as overrides to particular settings such as `deviceType`. Check the possible values to set in `ExampleConfig` in `scripts/utils.mjs` file for the full list.
```
Comment (willeastcott) [inline examples/README.md:60]:
```suggestion
Examples can also contain comments which allow you to define the default configuration for your examples as well as overrides to particular settings such as `deviceType`. Check the possible values to set in `ExampleConfig` in `scripts/utils.mjs` file for the full list.
```

File: examples/scripts/utils.mjs [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import fs from 'fs';
+
+/**
+ * @param {object} obj - The object.
+ * @returns {string} - The stringifiied object
```
Comment (willeastcott) [inline examples/scripts/utils.mjs:5]:
```suggestion
 * @returns {string} - The stringified object
```
